### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by chriskinsman**
Added support for specifying a proxy and passing it onto request.

### CODE REVIEW

File: lib/list.js [hunk 1]
```diff
@@ -11,9 +11,16 @@ function list (opts) {
   return new Promise(function (resolve, reject) {
     opts = R.clone(opts || {});
     validate(opts);
-    const url = buildUrl(opts);
 
-    request(url, opts.throttle)
+    let options = {
```
Comment (facundoolano) [inline lib/list.js]: make this a const, and add a `proxy: opts.proxy` instead of using a separated if. The same applies for the resto of the files.

**Comment by facundoolano** [comment]
Hi @chriskinsman, thanks for the PR, I just left you a code comment. 

Also, note the build failed in Travis because of style issues, you'll have to fix that before I can merge.

**Comment by chriskinsman** [comment]
All fixed up.

**Comment by facundoolano** [comment]
Thanks!

**Pull request merged**